internal/fields/validate.go

```diff
@@ -855,6 +857,22 @@ func isArrayOfObjects(val any) bool {
 	return false
 }
 
+func isFlattenedSubfield(key string, schema []FieldDefinition) bool {
+	for strings.Contains(key, ".") {
+		i := strings.LastIndex(key, ".")
+		key = key[:i]
+		ancestor := FindElementDefinition(key, schema)
+		if ancestor == nil {
+			continue
+		}
+		if ancestor.Type == "flattened" {
+			return true
+		}
```

Comment:
If it find an ancestor with a different type of "flattened" should be return false directly here?
I was thinking in the case of the ancestor being something like: `ancestor.Type = object` and `ancestor.ObjectTYpe == "keyword"`

Reply:
Not sure, there are packages defining mappings for subfields of flattened objects (for example [here](https://github.com/elastic/integrations/blob/2074f69f7681b1c2f04a03c228173ab425cd0283/packages/auditd_manager/data_stream/auditd/fields/fields.yml#L695)). This doesn't make a lot of sense as a mapping, but this is allowed now, and can be used for example to provide documentation.

We probably should not allow to define mappings under `flattened` fields, but this would be a different task, and checked in a different place. And we would need to provide something for cases where developers want to provide docs for flattened subfields.

Reply:
> Not sure, there are packages defining mappings for subfields of flattened objects (for example [here](https://github.com/elastic/integrations/blob/2074f69f7681b1c2f04a03c228173ab425cd0283/packages/auditd_manager/data_stream/auditd/fields/fields.yml#L695)). This doesn't make a lot of sense as a mapping, but this is allowed now, and can be used for example to provide documentation.

Ok, I didn't know about this kind of usage.

> We probably should not allow to define mappings under flattened fields, but this would be a different task, and checked in a different place. And we would need to provide something for cases where developers want to provide docs for flattened subfields.

Agreed!